broker/lib/fabrik/DBManager.js

```diff
@@ -214,7 +231,21 @@ class DBManager {
           logger.info(`MongoDB ${operation} request is complete. Check status for task id - ${taskId}`);
           this.director
             .pollTaskStatusTillComplete(taskId)
-            .then(response => this.dbCreateUpdateSucceeded(response, createIfNotPresent))
+            .then(response => {
+              return Promise.try(() => {
+                  if (createIfNotPresent) {
+                    //This is precaution to ensure that no previous bind resource exists in the event that
```

Comment:
better to update the resource if it exists. Currently you are handling only NotFound exception here, but due to some other issue if delete fails, operation will result in failure. Also since bind is being fired without checking, you need to handle NotFound, specifically. However if you had this as part of getResource then either create/update as part of storeBindingProperty it would make it more readable. 
Even this approach is fine, if you want to continue this, move this check at LN228. So that if this fails, unnecessarily you dont end up provisioning DB and you fail fast.

Reply:
I agree that readability will improve if this logic moves to storeBindingPropery. So if I understand correctly: In storeBindingProperty, based on output of getResource call (NotFound or successful)we will either create or update the bind resource. Can you please confirm this understanding?